plugin-server/src/main/ingestion-queues/ingest-event.ts

```diff
@@ -1,14 +1,14 @@
 import { PluginEvent } from '@posthog/plugin-scaffold'
 import * as Sentry from '@sentry/node'
 
-import { Hub, WorkerMethods } from '../../types'
+import { Hub, IngestionWorkerMethods, WorkerMethods } from '../../types'
 import { timeoutGuard } from '../../utils/db/utils'
 import { status } from '../../utils/status'
 import { Action } from './../../types'
 
 export async function ingestEvent(
     server: Hub,
-    workerMethods: WorkerMethods,
+    workerMethods: IngestionWorkerMethods,
     event: PluginEvent,
     checkAndPause?: () => void // pause incoming messages if we are slow in getting them out again
 ): Promise<void> {
```

Comment:
Maybe refactor this so it's similar to `onEvent` & move the event processing part (i.e. applying geoIp etc to be separate function)